rosidl_generator_c/src/string_functions.c

```diff
@@ -43,7 +44,7 @@ rosidl_generator_c__String__fini(rosidl_generator_c__String * str)
     str->size = 0;
     str->capacity = 0;
   } else {
-    /* ensure that data, szie and and capacity values are consistent */
+    /* ensure that data, size, and capacity values are consistent */
     assert(0 == str->size);
     assert(0 == str->capacity);
```

Comment:
If the rest of the code is using `assert`, then I guess you don't have to change it. The code in general doesn't handle errors very consistently.